lib/rules/jsx-handler-names.js

```diff
@@ -86,14 +105,27 @@ module.exports = {
 
     const checkLocal = !!configuration.checkLocalVariables;
 
+    const checkInlineFunction = !!configuration.checkInlineFunction;
+
     return {
       JSXAttribute(node) {
-        if (!node.value || !node.value.expression || (!checkLocal && !node.value.expression.object)) {
+        if (!node.value || !node.value.expression || (!checkLocal && (isInlineHandler(node)
+          ? !node.value.expression.body.callee.object
+          : !node.value.expression.object))) {
           return;
         }
 
+        if (checkInlineFunction && !isInlineHandler(node)) {
+          context.report({
+            node,
+            message: 'Expected handler function should be inline function.'
+          });
+        }
+
         const propKey = typeof node.name === 'object' ? node.name.name : node.name;
-        const propValue = context.getSourceCode().getText(node.value.expression).replace(/^this\.|.*::/, '');
+        const propValue = isInlineHandler(node)
+          ? isLocalInlineHandler(node)
+          : context.getSourceCode().getText(node.value.expression).replace(/^this\.|.*::/, '');
```

Comment:
I'm not familiar with how eslint rules work, so you should probably have another person review this just to be certain!

Reply:
This situation is to get the name of handler function.

1. If handler function is an inline function, you have to care whether it's local variable or not. [AST structure](https://github.com/yannickcr/eslint-plugin-react/pull/2761/files#r475780562)
2. If handler function is not an inline function, the name of handler function can be imported into regex. [line 128](https://github.com/yannickcr/eslint-plugin-react/pull/2761/files#diff-ccf82a24ec6f364517e4d1f4aed7a0a8R128)

Reply:
Ah got it - ok this looks fine to me!

Reply:
https://github.com/yannickcr/eslint-plugin-react/pull/2761#discussion_r476542714